### PR TITLE
[SPARK-25576][BUILD][BRANCH-2.2] Fix lint failure

### PR DESCRIPTION
**Pull request by samdvr**
## What changes were proposed in this pull request?

Line length fixes and 

## How was this patch tested?

Manually verified, but will ensure jenkins lint passes before merging 

Related Job:
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Compile/job/spark-branch-2.2-lint/913/console

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
Can you link the JIRA https://issues.apache.org/jira/browse/SPARK-25576 ? Please see https://spark.apache.org/contributing.html

**Comment by samdvr** [comment]
@HyukjinKwon linked the JIRA to this PR and tests have passed.

**Comment by dongjoon-hyun** [comment]
Thank you for your first contribution, @samdvr . Could you update the title, `[SPARK-25576][BUILD][BRANCH-2.2] Fix lint failure`?

File: core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java [hunk 1]
```diff
@@ -174,9 +174,12 @@ public void reset() {
     if (consumer != null) {
       consumer.freeArray(array);
       // the call to consumer.allocateArray may trigger a spill
-      // which in turn access this instance and eventually re-enter this method and try to free the array again.
-      // by setting the array to null and its length to 0 we effectively make the spill code-path a no-op.
-      // setting the array to null also indicates that it has already been de-allocated which prevents a double de-allocation in free().
+      // which in turn access this instance and eventually re-enter this method 
+      // and try to free the array again.
+      // by setting the array to null and its length to 0
```
Comment (dongjoon-hyun) [inline core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java]: `by` -> `By`?

File: core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java [hunk 2]
```diff
@@ -174,9 +174,12 @@ public void reset() {
...
       consumer.freeArray(array);
       // the call to consumer.allocateArray may trigger a spill
-      // which in turn access this instance and eventually re-enter this method and try to free the array again.
-      // by setting the array to null and its length to 0 we effectively make the spill code-path a no-op.
-      // setting the array to null also indicates that it has already been de-allocated which prevents a double de-allocation in free().
+      // which in turn access this instance and eventually re-enter this method 
+      // and try to free the array again.
+      // by setting the array to null and its length to 0 
+      // we effectively make the spill code-path a no-op.
+      // setting the array to null also indicates that it has already been
```
Comment (dongjoon-hyun) [inline core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java]: `setting` -> `Setting`?

**Comment by samdvr** [comment]
@dongjoon-hyun updated per your comments, thank you.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Actually, this issue aims to pass `dev/lint-java`. For now, this PR seems to be unable to pass the Jenkins. Could you check it by run `dev/lint-java` in your spark home directory?

```scala
$ dev/lint-java
exec: curl --progress-bar -L https://downloads.typesafe.com/zinc/0.3.11/zinc-0.3.11.tgz
######################################################################## 100.0%
exec: curl --progress-bar -L https://downloads.typesafe.com/scala/2.11.8/scala-2.11.8.tgz
######################################################################## 100.0%
Using `mvn` from path: /usr/local/bin/mvn
Checkstyle checks failed at following occurrences:
[ERROR] src/test/java/org/apache/spark/unsafe/types/UTF8StringSuite.java:[66] (misc) AvoidEscapedUnicodeCharacters: Unicode escape(s) usage should be avoided.
[ERROR] src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java:[177] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java:[179] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java:[181] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java:[473] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java:[478] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java:[485] (regexp) RegexpSingleline: No trailing whitespace allowed.
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java:[489] (sizes) LineLength: Line is longer than 100 characters (found 120).
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java:[491] (sizes) LineLength: Line is longer than 100 characters (found 114).
[ERROR] src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorterSuite.java:[186] (sizes) LineLength: Line is longer than 100 characters (found 116).
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
I made a PR to you, @samdvr . Could you merge that, https://github.com/samdvr/spark/pull/1 ?

**Comment by samdvr** [comment]
@dongjoon-hyun merged, thank you, did not know how to skip lint rule there.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @samdvr .
Merged to branch-2.2.

**Comment by dongjoon-hyun** [comment]
BTW, I chose the most official one, `Sam Davarnia <sdavarnia@apple.com>`. But, it seems that your GitHub is not happy with that. Next time, you had better be consistent in your commit message.
```
Lead-authored-by: Sam Davarnia <sdavarnia@apple.com>
Co-authored-by: Sam Davarnia <>
Co-authored-by: Dongjoon Hyun <dongjoon@apache.org>
Co-authored-by: Sam Davarnia <samdvr@users.noreply.github.com>
Signed-off-by: Dongjoon Hyun <dongjoon@apache.org>
```

**Comment by dongjoon-hyun** [comment]
Since it's merged into `branch-2.2`, could you close this PR? GitHub only closes the PRs against `master` branch. :)

**Comment by samdvr** [comment]
Thanks will do!

**Pull request closed without merging**